LLVM.Imports.Core.pas

```diff
@@ -2132,6 +2497,163 @@ procedure LLVMStopMultithreaded; cdecl; external CLLVMLibrary;
     @see llvm::llvm_is_multithreaded *)
 function LLVMIsMultithreaded: LongBool; cdecl; external CLLVMLibrary;
 
+// Added by Max 13/09/2019 17:22:45
+(** Return the function this instruction belongs to.
+ *
+ * Note: it is undefined behavior to call this on an instruction not
+ * currently inserted into a function.  see : const Function *llvm::getFunction()
+ *)
+function LLVMGetInstructionFunc(Inst: TLLVMValueRef):TLLVMValueRef;
+function LLVMGetUsesVal(val : TLLVMValueRef): TArray<TLLVMValueRef> ;
+function LLVMGetNextNonDebugInstruction(I: TLLVMValueRef): TLLVMValueRef;
+function LLVMGetPrevNonDebugInstruction(I: TLLVMValueRef): TLLVMValueRef;
+function LLVMDumpValueToStr(v:TLLVMValueRef) : AnsiString;
```

Comment:
I prefer to use string instead of AnsiString as returntypes

Reply:
why?
LLVMPrintValueToString (v) returns pansichar for
what reason to make a superfluous conversion?

``function LLVMDumpValueToStr(v:TLLVMValueRef) : AnsiString;
var
  pDump : PAnsiChar;
begin
    pDump := LLVMPrintValueToString(v);
    try
      Result := AnsiString(pDump) ;
    finally
      LLVMDisposeMessage(pDump);
    end;
end;``

Reply:
nothing i'll enforce here but usually people work with string in their code and you end up writing string(functioncall) to avoid "implicit conversion" warnings.